zone/client.cpp

```diff
@@ -10129,3 +10138,472 @@ void Client::SetAFK(uint8 afk_flag) {
+		distance += delta;
+		delta = (z - (-12.8f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		delta = (x - (-153.8f));
+		delta *= delta;
+		distance = delta;
+		delta = (y - (-30.3f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (8.2f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		break;
+	}
+	case 9:
+	{
+		float delta = (x - (-682.5f));
+		delta *= delta;
+		float distance = delta;
+		delta = (y - (147.0f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (-9.9f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		delta = (x - (-655.4f));
+		delta *= delta;
+		distance = delta;
+		delta = (y - (10.5f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (-51.8f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		break;
+	}
+	case 62:
+	case 75:
+	case 114:
+	case 209:
+	{
+		//The portals are so common in paineel/felwitheb that checking
+		//distances wouldn't be worth it cause unless you're porting to the
+		//start field you're going to be triggering this and that's a level of
+		//accuracy I'm willing to sacrifice
+		return true;
+		break;
+	}
+
+	case 24:
+	{
+		float delta = (x - (-183.0f));
+		delta *= delta;
+		float distance = delta;
+		delta = (y - (-773.3f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (54.1f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		delta = (x - (-8.8f));
+		delta *= delta;
+		distance = delta;
+		delta = (y - (-394.1f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (41.1f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		delta = (x - (-310.3f));
+		delta *= delta;
+		distance = delta;
+		delta = (y - (-1411.6f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (-42.8f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		delta = (x - (-183.1f));
+		delta *= delta;
+		distance = delta;
+		delta = (y - (-1409.8f));
+		delta *= delta;
+		distance += delta;
+		delta = (z - (37.1f));
+		delta *= delta;
+		distance += delta;
+
+		if (distance < max_dist)
+			return true;
+
+		break;
+	}
+
+	case 110:
+	case 34:
+	case 96:
+	case 93:
+	case 68:
+	case 84:
+	{
+		if (GetBoatID() != 0)
+			return true;
+		break;
+	}
+	default:
+		break;
+	}
+	return false;
+}
+
+void Client::CheatDetected(CheatTypes CheatType, glm::vec3 from, glm::vec3 to)
+{
+	switch (CheatType)
+	{
+	case MQWarp:
+		if (RuleB(Zone, EnableMQWarpDetector)
+			&& ((Admin() < RuleI(Zone, MQWarpExemptStatus)
+				|| (RuleI(Zone, MQWarpExemptStatus)) == -1)))
+		{
+			if (!RuleB(Zone, MQDetectionSilentReport))
+				Message(13, "Large warp detected.");
+			char hString[250];
+			sprintf(hString, "/MQWarp(LWD) with location from: %.2f, %.2f, %.2f", from.x, from.y, from.z); 
```

Comment:
Please convert to `fmt::format`

Please don't use acronyms in the messages, just make it clear what it is that the message is relevant to

```cpp
std::string hacker_message = fmt::format("/MQWarp (large warp detection) with location from x [{:.2f}] y [{:.2f}] z [{:.2f}]");
```

When passed into the existing DetectionFlag signature it will have to be cast into an old C string

Please do this for all of the other messages as well.

Since we're already logging this let's make sure that we send these to `LogCheat` as well so we have the debugging already hooked up for the next developer who picks this up